File: integration-env/Dockerfile.c

FROM ubuntu:22.04
# Installation de clang (demandé par votre Makefile) et libmicrohttpd
RUN apt-get update && apt-get install -y \
    clang \
    make \
    libmicrohttpd-dev \
    curl \
    && rm -rf /var/lib/apt/lists/*

WORKDIR /app
COPY . .

# Compilation du wrapper web
RUN clang -Wall -o auth_service web_wrapper.c -lmicrohttpd

EXPOSE 8082

# Lancement
CMD ["./auth_service", "8082"]

File: integration-env/Dockerfile.java

FROM amazoncorretto:17-alpine
WORKDIR /app
# Installation de curl pour le test de santé (healthcheck)
RUN apk add --no-cache curl
# Copie des sources
COPY . .
# Compilation directe du fichier unique et de son wrapper web
RUN javac src/main/java/com/kimvieware/auth/AuthService.java src/main/java/com/kimvieware/auth/WebAuthService.java
# Exposition du port
EXPOSE 8081
# Lancement du service web
CMD ["java", "-cp", "src/main/java", "com.kimvieware.auth.WebAuthService"]
